tv-shows/src/app/components/shared/StarsRating/StarsRating.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import { Box, Flex } from "@chakra-ui/react";
+import { useState } from "react";
+
+interface IStarsRatingProps {
+  canInteract: boolean;
+  rating: number;
+  setSelectedRating?: (rating: number) => void;
+}
+
+function getStarsColor(
+  canInteract: boolean,
+  rating: number,
+  hoveredRating: number | null,
+  index: number
+) {
+  if (canInteract) {
+    if (hoveredRating !== null) {
+      return hoveredRating > index ? "yellow.400" : "white";
+    }
+    return rating > index ? "yellow.400" : "white";
+  }
+  return rating > index ? "yellow.400" : "white";
+}
+
+export const StarsRating = ({
+  rating,
+  canInteract = false,
+  setSelectedRating,
+}: IStarsRatingProps) => {
+  const [hoveredRating, setHoveredRating] = useState<number | null>(null);
+
+  return (
+    <Flex gap="2" my={4}>
+      {Array.from({ length: 5 }, (_, index) => (
```

Comment:
Everybody keeps using `_` for the unused variable in this function because they saw it on SOverflow. I tell everyone to use the full variable name because I am petty.